### DISCUSSION TITLE
How can a beginner start with Wezterm's multiplexing feature?

### DISCUSSION THREAD

**Discussion by ziontee113** [wezterm/wezterm#2923] [Q&A] [answered]
Hi, I'm trying to replace tmux with Wezterm due to the lack of support for escape key combinations (which Wezterm do perfectly). I'm just quite lost on how I can start a new session with Wezterm.

Is there any command equivalent to just running `tmux`? (start new session, create 1 new window).

All I want is to close my terminal without losing everything inside of it. I wish to close the terminal and can get back the still running session later.

Thank you very much :pray:

**Comment by wez** [comment, marked as answer]
Hi!

Take a look at the configuration at the top of this issue:
* https://github.com/wez/wezterm/issues/2933

that will cause wezterm to use a local unix domain multiplexer by default and use that.

Note that that issue is open for a problem on X11 and Wayland systems with how the default .desktop file (that is what makes wezterm appear in application launchers) overrides the default arguments, and the rest of the issue is discussing that problem and workarounds.

**Reply by ziontee113** [reply]
Thank you for the reply :+1:

**Reply by robrecord** [reply]
Thanks @wez - Is this still the recommended way? I heard this was no longer the recommended method for reasons of performance, or did I misunderstand?

Also the colorscheme is missing when I do this - I haven't figured out how to make the mux server accept my local configuration file. Could this also be added to the docs? Happy to make a PR as soon as I can understand how it's done.

**Reply by wez** [reply]
re: colors, see the note in this section: https://wezfurlong.org/wezterm/config/appearance/#color-scheme

re: multiplexing, I personally do not use local unix domain multiplexing. The performance is much faster when bypassing that and just use the multiplexing that is built right into the gui, rather than making it work harder to "remote" on a local connection.  However, if you truly need to be able to quit the gui and re-attach to your local mux, then using unix domain multiplexing is the way to do that.

**Reply by robrecord** [reply]
Thanks Wez, I understand. I am looking for tmux-like features such as having multiple sessions (workspaces) open in different windows. I’ll keep an eye out for future developments in that regard. 

Re the colours, yes I’ve seen that comment but it’s not clear what should be done when using a local unix domain. I presume the mux-server looks for the config in the same place as the gui? It’s there in my .config/wezterm, so shouldn’t the mux server pick it up from there?  I’ve also tried independently starting wezterm-mux-server from the command line with —config-file (before wezterm starts) but it doesn’t seem to make a difference.

**Reply by wez** [reply]
@robrecord please file an issue for that so I can dig in and follow up

**Reply by valsuo** [reply]
" re: multiplexing, I personally do not use local unix domain multiplexing. The performance is much faster when bypassing that and just use the multiplexing that is built right into the gui, rather than making it work harder to "remote" on a local connection. However, if you truly need to be able to quit the gui and re-attach to your local mux, then using unix domain multiplexing is the way to do that."

what does local multiplexing mean?  splits and tabs?

**Reply by wez** [reply]
multiplexing is running more than one pane in a given process, so any combination of additional windows, tabs and panes is exercising the multiplexing function built into the local wezterm gui process.

**Reply by theAnuragMishra** [reply]
> re: colors, see the note in this section: https://wezfurlong.org/wezterm/config/appearance/#color-scheme
> 
> re: multiplexing, I personally do not use local unix domain multiplexing. The performance is much faster when bypassing that and just use the multiplexing that is built right into the gui, rather than making it work harder to "remote" on a local connection. However, if you truly need to be able to quit the gui and re-attach to your local mux, then using unix domain multiplexing is the way to do that.

Hi @wez, what do you mean by unix domain multiplexing? I am just starting with wezterm and i want to be able to create named panes&tabs configurations such that i can start wezterm with that number of tabs opened split in defined ways in different panes. i dont necessarily want that when i close the terminal running processes persist as in tmux. 
to be precise: i just want that when i have to work on the project next time i have to split the tab into panes and resize them again.

**Comment by nkolatsis** [comment]
@wez It would be good to have a vim tutor style document for multiplexing with wezterm. As a beginner switching from vscode, I'm basically forced to use tmux just because there is a lot of documentation out there on how to get up and running.

**Reply by xpufx** [reply]
> @wez It would be good to have a vim tutor style document for multiplexing with wezterm. As a beginner switching from vscode, I'm basically forced to use tmux just because there is a lot of documentation out there on how to get up and running.

As a basic flow on a remote host just use the command palette to pick the SSHMUX connection for one of your hosts (where wezterm is also instealled on the other side) then use it like you would use wezterm normally. Your pane shortcuts etc will work the same. When done and you want to keep that sessions open, type DETACH in the command palette and detach the sshmux sessions you're connected to.  If you have only this one muxed session, you can just repeat the process to implicitly connect to the same mux  ssh domain and get your session back.

 You  might have meant something entirely more encompassing but this is the sum total of MY mux usage and I love it. (except for some issues with super active and interactive panes like you get with conding CLIs).
